worldedit-core/src/main/java/com/sk89q/worldedit/function/operation/Operation.java

```diff
@@ -56,7 +60,22 @@
      * of the operation.
      *
      * @param messages The list to add messages to
+     * @deprecated Will be removed in WorldEdit 8.0 - use the Component variant
      */
-    void addStatusMessages(List<String> messages);
+    @Deprecated
+    default void addStatusMessages(List<String> messages) {
+    }
 
+    /**
+     * Gets an iterable of messages that describe the current status of the
+     * operation.
+     *
+     * @return The status messages
+     */
+    default Iterable<Component> getStatusMessages() {
+        // TODO Remove legacy code WorldEdit 8.0.0
```

Comment:
Can we add a one-time-deprecation-warning here? Just something to show it in the server logs of users a little bit too.

e.g. "An operation is using the old status message API: \<class name>", and just add the class to a set to skip the message on next calls.